vars/standardReleasePipelineWithGenericTrigger.groovy

```diff
@@ -37,21 +38,51 @@ void call(Map args = [:], Closure body) {
             GenericTrigger(
                 genericVariables: [
                     [key: 'ref', value: (args.jsonValue ?: '$.release.tag_name')],
+                    [key: 'action', value: '$.action'],
                     [key: 'isDraft', value: '$.release.draft'],
-                    [key: 'release_url', value: '$.release.url']
+                    [key: 'release_url', value: '$.release.url'],
+                    [key: 'assets_url', value: '$.release.assets_url']
                 ],
                 tokenCredentialId: args.tokenIdCredential,
                 causeString: args.causeString ?: 'A tag was cut on GitHub repository causing this workflow to run',
                 printContributedVariables: false,
                 printPostContent: false,
-                regexpFilterText: (args.regexpFilterText ?: '$isDraft'),
-                regexpFilterExpression: (args.regexpFilterExpression ?: 'true')
+                regexpFilterText: (args.regexpFilterText ?: '$isDraft $action'),
+                regexpFilterExpression: (args.regexpFilterExpression ?: '^true created$')
             )
         }
         environment {
             tag = "$ref"
         }
         stages {
+            stage('Download artifacts') {
+                when {
+                    expression {
+                        return args.downloadReleaseAsset
+                    }
+                }
+                steps {
+                    script {
+                        if (args.downloadReleaseAsset && "$assets_url" != '') {
+                            withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
+                                String assets = sh(
```

Comment:
A bit confused what is the need for assets if value can be retrieved from assetUrl?

Thanks

Reply:
assetUrl is just the api to get all the assets from. Something like: "assets_url": "https://api.github.com/repos/gaiksaya/opensearch-py-ml/releases/82615186/assets"
When you curl this api, this is when you get exact assets like here https://github.com/opensearch-project/opensearch-build-libraries/pull/32/files#diff-cb975467ca0159d0c434e26ed12f26fc453d36cd692d24cf47812eb137f3574bR26

Reply:
Got it thanks.